zmon_worker_monitor/builtins/plugins/datapipeline.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import boto3
+import logging
+import requests
+
+
+from zmon_worker_monitor.zmon_worker.errors import CheckError
+from zmon_worker_monitor.adapters.ifunctionfactory_plugin import IFunctionFactoryPlugin, propartial
+
+
+logging.getLogger('botocore').setLevel(logging.WARN)
+
+
+class DataPipelineWrapperFactory(IFunctionFactoryPlugin):
+    def __init__(self):
+        super(DataPipelineWrapperFactory, self).__init__()
+
+    def configure(self, conf):
+        return
+
+    def create(self, factory_ctx):
+        """
+        Automatically called to create the check function's object
+        :param factory_ctx: (dict) names available for Function instantiation
+        :return: an object that implements a check function
+        """
+        return propartial(DataPipelineWrapper, region=factory_ctx.get('entity').get('region', None))
+
+
+def get_region():
+    r = requests.get('http://169.254.169.254/latest/dynamic/instance-identity/document', timeout=3)
+    return r.json()['region']
+
+
+# create a dict of keys from a list of dicts
+def create_dict_from_list_of_fields(fields):
+    fields_dict = {}
+    for field in fields:
+        fields_dict[str(field['key'])] = str(field['stringValue'])
+
+    return fields_dict
+
+
+class DataPipelineWrapper(object):
+    def __init__(self, region=None):
+        if not region:
+            region = get_region()
+        self.__client = boto3.client('datapipeline', region_name=region)
+
+    """
+        The function takes as parameter a list of pipeline IDs as strings and returns a list of Maps
+        with the following signature:
+        [{pipelineId: {
+            propertyKey1: Value1,
+            propertyKey2: Value2
+            ...
+            },
+        ...]
+    """
+    def get_details(self, pipeline_ids):
+
+        if type(pipeline_ids) is str:
+            pipeline_ids = [pipeline_ids]
+        else:
+            if not type(pipeline_ids) is list:
```

Comment:
More pythonic object identity check is using ``is not`` instead of ``not ... is ...``.

Also using ``isinstance(pipeline_ids, list)`` is considered more accurate (e.g. in the case of sub-classing) but it might not be strictly needed here :)